cranelift-codegen/meta/src/isa/x86/instructions.rs

```diff
@@ -307,5 +307,74 @@ pub fn define(
         .operands_out(vec![a]),
     );
 
+    let x = &operand("x", TxN);
+    let y = &operand_doc("y", &TxN.lane_of(), "New lane value");
+    let a = &operand("a", TxN);
+
+    ig.push(
+        Inst::new(
+            "x86_pinsr",
+            r#"
+        Insert ``y`` into ``x`` at lane ``Idx``.
+        The lane index, ``Idx``, is an immediate value, not an SSA value. It
+        must indicate a valid lane index for the type of ``x``.
+        "#,
+        )
+        .operands_in(vec![x, Idx, y])
```

Comment:
Similarly to what you've done for insertps below, should this be restricted to "Int x N" input / output operands?

Reply:
But this could be used for either integers or booleans, right?

Reply:
That's right; is TxN only limited to int or bools? (if so please ignore me)

Reply:
I think it's ints/floats/bools; you're saying I should create an `IBxN` just for bools/ints then since that's all this instruction should handle?

Reply:
Yes, it would help catch more type errors.